hibernate-core/src/test/java/org/hibernate/orm/test/batch/BatchOptimisticLockingTest.java

```diff
@@ -107,10 +109,19 @@ public void testBatchAndOptimisticLocking() {
 			}
 			else {
 				assertEquals( OptimisticLockException.class, expected.getClass() );
-				assertTrue(
-						expected.getMessage()
-								.startsWith("Batch update returned unexpected row count from update 1 (expected row count 1 but was 0) [update Person set name=?,version=? where id=? and version=?]")
-				);
+
+				if ( getDialect() instanceof MariaDBDialect && expected.getCause() instanceof LockAcquisitionException) {
```

Comment:
I guess this will fail with the old exception on older MariaDB versions even with the new driver? If so, this if condition needs an additional version check.

Reply:
it will works, but has been rewritten to be clear